### DISCUSSION TITLE
Syncing Play Progress from Trakt to Plex

### DISCUSSION THREAD

**Discussion by mroodev** [Taxel/PlexTraktSync#1885] [Q&A] [answered]
Hello there!

I've been using PlexTraktSync to keep my watch statuses synchronized between Plex and Trakt, which works perfectly. However, I'm seeking a solution to also sync the **playback progress** for movies and TV shows from Trakt _into_ Plex (so scrobbling is not what I am looking for).
My TV setup involves Kodi with the Trakt plugin, where my playback progress is accurately updated on Trakt (confirmed via [Trakt Playback Progress Manager](https://sharkykh.github.io/tppm/)).

I would like to continue watching content on Plex right where I left off in Kodi, ensuring a seamless transition between platforms without the need to manually remember where I paused (it's of course fine if I have to trigger a sync first).
When addons for Plex were still a thing, this was possible and worked great. But since that is gone, I have no idea how to achieve that anymore.

Does anyone have any insights or methods to achieve this? In summary I'm looking for ways to import my playback progress from Trakt directly into Plex, focusing on the current play position rather than just the watched status.

Any guidance or suggestions you could share would be immensely appreciated. Thank you in advance for your help!

**Comment by glensc** [comment]
find from plex python api, do they offer any methods to retrieve and set watch status point, if they do, then maybe it's doable to sync also play status.

and what plex addon you refer? please link.

**Reply by JonnyWong16** [reply]
`updateProgress(ms)`

https://python-plexapi.readthedocs.io/en/latest/modules/base.html#plexapi.base.Playable.updateProgress

**Reply by glensc** [reply]
@JonnyWong16 and is the same data available for read from plex?

**Reply by JonnyWong16** [reply]
It's `viewOffset`.

https://python-plexapi.readthedocs.io/en/latest/modules/video.html#plexapi.video.Movie

**Comment by glensc** [comment, marked as answer]
Started something:
- https://github.com/Taxel/PlexTraktSync/pull/1889

currently, it's able to sync from trakt to plex. I have ran myself the code only with `--dry-run`

**Reply by glensc** [reply]
@ItsVictory test it and give feedback!

**Reply by mroodev** [reply]
sorry I'm a bit late with my reply: just did a basic test with the main-branch & the newly added config setting and it synced the playback progress times from trakt to plex as expected - thank you so much, that's awesome!
going to test some more and definitely will give feedback/open an issue if I'll find something.
